packages/rest/src/rest.server.ts

```diff
@@ -272,14 +290,24 @@ export class RestServer extends Context implements Server, HttpServerLike {
     const mapping = this.config.openApiSpec!.endpointMapping!;
     // Serving OpenAPI spec
     for (const p in mapping) {
-      this._expressApp.get(p, (req, res) =>
-        this._serveOpenApiSpec(req, res, mapping[p]),
+      expressCtx.middleware(
+        (req, res) => this._serveOpenApiSpec(req, res, mapping[p]),
```

Comment:
I think our REST layer should be providing enough features to allow us to implement `_serverOpenApiSpec` routes as regular LB4 routes with `x-visibility: undocumented`. Instead of relying on your new extension, can we rework this part to use existing LB4 means?

Reply:
Ah, it looks like it was an intentional decision to implement these routes at REST API layer, bypassing the sequence.

https://github.com/strongloop/loopback-next/blob/c3d800700b253e97316fd0871641ea80fcb457f3/packages/rest/src/rest.server.ts#L265-L271

Reply:
Uh oh, this is even more complex than I thought. It turns out that OpenAPI endpoints like `/openapi.json` do not honor `basePath` configuration set at server level.

Reply:
You can see the RouteEntry-based implementation of `/openapi.json` here: 2b5be221

Reply:
> Uh oh, this is even more complex than I thought. It turns out that OpenAPI endpoints like `/openapi.json` do not honor `basePath` configuration set at server level.

This problem is already tracked here: https://github.com/strongloop/loopback-next/issues/2329